File: android/src/main/java/io/openim/flutter_openim_sdk/manager/ConversationManager.java
package io.openim.flutter_openim_sdk.manager;

import io.flutter.plugin.common.MethodCall;
import io.flutter.plugin.common.MethodChannel;
import io.openim.flutter_openim_sdk.listener.OnBaseListener;
import io.openim.flutter_openim_sdk.listener.OnConversationListener;
import io.openim.flutter_openim_sdk.util.CommonUtil;
import open_im_sdk.Open_im_sdk;

public class ConversationManager extends BaseManager {

    public void setConversationListener(MethodCall methodCall, MethodChannel.Result result) {
        Open_im_sdk.setConversationListener(new OnConversationListener());

        result.success(null);
    }


    public void getAllConversationList(MethodCall methodCall, MethodChannel.Result result) {
        Open_im_sdk.getAllConversationList(
                new OnBaseListener(result, methodCall),
                value(methodCall, "operationID")
        );
    }

    public void getConversationListSplit(MethodCall methodCall, MethodChannel.Result result) {
        Open_im_sdk.getConversationListSplit(
                new OnBaseListener(result, methodCall),
                value(methodCall, "operationID"),
                int2long(methodCall, "offset"),
                int2long(methodCall, "count")
        );
    }

    public void getOneConversation(MethodCall methodCall, MethodChannel.Result result) {
        Open_im_sdk.getOneConversation(
                new OnBaseListener(result, methodCall),
                value(methodCall, "operationID"),
                value(methodCall, "sessionType"),
                value(methodCall, "sourceID")
        );
    }

    public void getMultipleConversation(MethodCall methodCall, MethodChannel.Result result) {
        Open_im_sdk.getMultipleConversation(
                new OnBaseListener(result, methodCall),
                value(methodCall, "operationID"),
                jsonValue(methodCall, "conversationIDList")
        );
    }


    public void setConversationDraft(MethodCall methodCall, MethodChannel.Result result) {
        Open_im_sdk.setConversationDraft(
                new OnBaseListener(result, methodCall),
                value(methodCall, "operationID"),
                value(methodCall, "conversationID"),
                value(methodCall, "draftText")
        );
    }

    public void hideConversation(MethodCall methodCall, MethodChannel.Result result) {
        Open_im_sdk.hideConversation(
                new OnBaseListener(result, methodCall),
                value(methodCall, "operationID"),
                value(methodCall, "conversationID")
        );
    }

    public void markConversationMessageAsRead(MethodCall methodCall, MethodChannel.Result result) {
        Open_im_sdk.markConversationMessageAsRead(
                new OnBaseListener(result, methodCall),
                value(methodCall, "operationID"),
                value(methodCall, "conversationID")
        );
    }

    public void getTotalUnreadMsgCount(MethodCall methodCall, MethodChannel.Result result) {
        Open_im_sdk.getTotalUnreadMsgCount(
                new OnBaseListener(result, methodCall),
                value(methodCall, "operationID")
        );
    }

    public void getConversationIDBySessionType(MethodCall methodCall, MethodChannel.Result result) {
        CommonUtil.runMainThreadReturn(result, Open_im_sdk.getConversationIDBySessionType(
                value(methodCall, "operationID"),
                value(methodCall, "sourceID"),
                int2long(methodCall, "sessionType")));
    }

    public void clearConversationAndDeleteAllMsg(MethodCall methodCall, MethodChannel.Result result) {
        Open_im_sdk.clearConversationAndDeleteAllMsg(
                new OnBaseListener(result, methodCall),
                value(methodCall, "operationID"),
                value(methodCall, "conversationID")
        );
    }

    public void deleteConversationAndDeleteAllMsg(MethodCall methodCall, MethodChannel.Result result) {
        Open_im_sdk.deleteConversationAndDeleteAllMsg(
                new OnBaseListener(result, methodCall),
                value(methodCall, "operationID"),
                value(methodCall, "conversationID")
        );
    }

    public void getAtAllTag(MethodCall methodCall, MethodChannel.Result result) {
        CommonUtil.runMainThreadReturn(result, Open_im_sdk.getAtAllTag(value(methodCall, "operationID")));
    }

    public void hideAllConversations(MethodCall methodCall, MethodChannel.Result result) {
        Open_im_sdk.hideAllConversations(
                new OnBaseListener(result, methodCall),
                value(methodCall, "operationID")
        );
    }

    public  void searchConversation(MethodCall methodCall, MethodChannel.Result result) {
        Open_im_sdk.searchConversation(
                new OnBaseListener(result, methodCall),
                value(methodCall, "operationID"),
                value(methodCall, "name")
        );
    }

    public void changeInputStates(MethodCall methodCall, MethodChannel.Result result) {
        Open_im_sdk.changeInputStates(
                new OnBaseListener(result, methodCall),
                value(methodCall, "operationID"),
                value(methodCall, "conversationID"),
                value(methodCall, "focus")
        );
    }

    public void getInputStates(MethodCall methodCall, MethodChannel.Result result) {
        Open_im_sdk.getInputStates(
                new OnBaseListener(result, methodCall),
                value(methodCall, "operationID"),
                value(methodCall, "conversationID"),
                value(methodCall, "userID")
        );
    }

    public void setConversation(MethodCall methodCall, MethodChannel.Result result) {
        Open_im_sdk.setConversation(
                new OnBaseListener(result, methodCall),
                value(methodCall, "operationID"),
                value(methodCall, "conversationID"),
                jsonValue(methodCall, "req")
        );
    }


    public void searchConversations(MethodCall methodCall, MethodChannel.Result result) {
        Open_im_sdk.searchConversation(
                new OnBaseListener(result, methodCall),
                value(methodCall, "operationID"),
                value(methodCall, "name")
        );
    }
}

File: ios/Classes/Module/ConversationManager.swift
import Foundation
import OpenIMCore

public class ConversationManager: BaseServiceManager {

    public override func registerHandlers() {
        super.registerHandlers()
        self["changeInputStates"] = changeInputStates
        self["clearConversationAndDeleteAllMsg"] = clearConversationAndDeleteAllMsg
        self["deleteConversationAndDeleteAllMsg"] = deleteConversationAndDeleteAllMsg
        self["getAllConversationList"] = getAllConversationList
        self["getAtAllTag"] = getAtAllTag
        self["getConversationIDBySessionType"] = getConversationIDBySessionType
        self["getConversationListSplit"] = getConversationListSplit
        self["getInputStates"] = getInputStates
        self["getMultipleConversation"] = getMultipleConversation
        self["getOneConversation"] = getOneConversation
        self["getTotalUnreadMsgCount"] = getTotalUnreadMsgCount
        self["hideAllConversations"] = hideAllConversations
        self["hideConversation"] = hideConversation
        self["markConversationMessageAsRead"] = markConversationMessageAsRead
        self["searchConversation"] = searchConversation
        self["setConversationDraft"] = setConversationDraft
        self["setConversationListener"] = setConversationListener
        self["setConversation"] = setConversation
        self["searchConversations"] = searchConversations
    }

    func changeInputStates(methodCall: FlutterMethodCall, result: @escaping FlutterResult) {
        Open_im_sdkChangeInputStates(BaseCallback(result: result), methodCall[string: "operationID"], methodCall[string: "conversationID"], methodCall[bool: "focus"])
    }

    func clearConversationAndDeleteAllMsg(methodCall: FlutterMethodCall, result: @escaping FlutterResult) {
        Open_im_sdkClearConversationAndDeleteAllMsg(BaseCallback(result: result), methodCall[string: "operationID"], methodCall[string: "conversationID"])
    }

    func deleteConversationAndDeleteAllMsg(methodCall: FlutterMethodCall, result: @escaping FlutterResult) {
        Open_im_sdkDeleteConversationAndDeleteAllMsg(BaseCallback(result: result), methodCall[string: "operationID"], methodCall[string: "conversationID"])
    }

    func getAllConversationList(methodCall: FlutterMethodCall, result: @escaping FlutterResult) {
        Open_im_sdkGetAllConversationList(BaseCallback(result: result), methodCall[string: "operationID"])
    }

    func getAtAllTag(methodCall: FlutterMethodCall, result: @escaping FlutterResult) {
        callBack(result, Open_im_sdkGetAtAllTag(methodCall[string: "operationID"]))
    }

    func getConversationIDBySessionType(methodCall: FlutterMethodCall, result: @escaping FlutterResult) {
        let conversationID = Open_im_sdkGetConversationIDBySessionType(methodCall[string: "operationID"], methodCall[string: "sourceID"], methodCall[int: "sessionType"])
        callBack(result, conversationID)
    }

    func getConversationListSplit(methodCall: FlutterMethodCall, result: @escaping FlutterResult) {
        Open_im_sdkGetConversationListSplit(BaseCallback(result: result), methodCall[string: "operationID"], methodCall[int: "offset"], methodCall[int: "count"])
    }

    func getInputStates(methodCall: FlutterMethodCall, result: @escaping FlutterResult) {
        Open_im_sdkGetInputStates(BaseCallback(result: result), methodCall[string: "operationID"], methodCall[string: "conversationID"], methodCall[string: "userID"])
    }

    func getMultipleConversation(methodCall: FlutterMethodCall, result: @escaping FlutterResult) {
        Open_im_sdkGetMultipleConversation(BaseCallback(result: result), methodCall[string: "operationID"], methodCall[jsonString: "conversationIDList"])
    }

    func getOneConversation(methodCall: FlutterMethodCall, result: @escaping FlutterResult) {
        Open_im_sdkGetOneConversation(BaseCallback(result: result), methodCall[string: "operationID"], methodCall[int32: "sessionType"], methodCall[string: "sourceID"])
    }

    func getTotalUnreadMsgCount(methodCall: FlutterMethodCall, result: @escaping FlutterResult) {
        Open_im_sdkGetTotalUnreadMsgCount(BaseCallback(result: result), methodCall[string: "operationID"])
    }

    func hideAllConversations(methodCall: FlutterMethodCall, result: @escaping FlutterResult) {
        Open_im_sdkHideAllConversations(BaseCallback(result: result), methodCall[string: "operationID"])
    }

    func hideConversation(methodCall: FlutterMethodCall, result: @escaping FlutterResult) {
        Open_im_sdkHideConversation(BaseCallback(result: result), methodCall[string: "operationID"], methodCall[string: "conversationID"])
    }

    func markConversationMessageAsRead(methodCall: FlutterMethodCall, result: @escaping FlutterResult) {
        Open_im_sdkMarkConversationMessageAsRead(BaseCallback(result: result), methodCall[string: "operationID"], methodCall[string: "conversationID"])
    }

    func searchConversation(methodCall: FlutterMethodCall, result: @escaping FlutterResult) {
        Open_im_sdkSearchConversation(BaseCallback(result: result), methodCall[string: "operationID"], methodCall[string: "name"])
    }

    func setConversationDraft(methodCall: FlutterMethodCall, result: @escaping FlutterResult) {
        Open_im_sdkSetConversationDraft(BaseCallback(result: result), methodCall[string: "operationID"], methodCall[string: "conversationID"], methodCall[string: "draftText"])
    }

    func setConversationListener(methodCall: FlutterMethodCall, result: @escaping FlutterResult) {
        Open_im_sdkSetConversationListener(ConversationListener(channel: channel))
        callBack(result)
    }

    func setConversation(methodCall: FlutterMethodCall, result: @escaping FlutterResult) {
        Open_im_sdkSetConversation(BaseCallback(result: result), methodCall[string: "operationID"], methodCall[string: "conversationID"], methodCall[jsonString: "req"])
    }

    func searchConversations(methodCall: FlutterMethodCall, result: @escaping FlutterResult) {
        Open_im_sdkSearchConversation(BaseCallback(result: result), methodCall[string: "operationID"], methodCall[string: "name"])
    }
}

public class ConversationListener: NSObject, Open_im_sdk_callbackOnConversationListenerProtocol {

    private let channel: FlutterMethodChannel

    init(channel: FlutterMethodChannel) {
        self.channel = channel
    }

    public func onConversationChanged(_ conversationList: String?) {
        CommonUtil.emitEvent(channel: channel, method: "conversationListener", type: "onConversationChanged", errCode: nil, errMsg: nil, data: conversationList)
    }

    public func onConversationUserInputStatusChanged(_ change: String?) {
        CommonUtil.emitEvent(channel: channel, method: "conversationListener", type: "onConversationUserInputStatusChanged", errCode: nil, errMsg: nil, data: change)
    }

    public func onNewConversation(_ conversationList: String?) {
        CommonUtil.emitEvent(channel: channel, method: "conversationListener", type: "onNewConversation", errCode: nil, errMsg: nil, data: conversationList)
    }

    public func onSyncServerFailed(_ reinstalled: Bool) {
        CommonUtil.emitEvent(channel: channel, method: "conversationListener", type: "onSyncServerFailed", errCode: nil, errMsg: nil, data: reinstalled)
    }

    public func onSyncServerFinish(_ reinstalled: Bool) {
        CommonUtil.emitEvent(channel: channel, method: "conversationListener", type: "onSyncServerFinish", errCode: nil, errMsg: nil, data: reinstalled)
    }

    public func onSyncServerProgress(_ progress: Int) {
        CommonUtil.emitEvent(channel: channel, method: "conversationListener", type: "onSyncServerProgress", errCode: nil, errMsg: nil, data: progress)
    }

    public func onSyncServerStart(_ reinstalled: Bool) {
        CommonUtil.emitEvent(channel: channel, method: "conversationListener", type: "onSyncServerStart", errCode: nil, errMsg: nil, data: reinstalled)
    }

    public func onTotalUnreadMessageCountChanged(_ totalUnreadCount: Int32) {
        CommonUtil.emitEvent(channel: channel, method: "conversationListener", type: "onTotalUnreadMessageCountChanged", errCode: nil, errMsg: nil, data: totalUnreadCount)
    }
}
